SportsHub/UnitTests/Services/TestAuthenticationService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Moq;
+using SportsHub.AppService.Authentication;
+using SportsHub.AppService.Services;
+using SportsHub.Domain.Models;
+using UnitTests.MockData;
+using Xunit;
+
+namespace UnitTests.Services;
+
+public class TestAuthenticationService
+{
+    private readonly Mock<IUserService> _userService;
+    private readonly IAuthenticationService _authentication;
+
+    public TestAuthenticationService()
+    {
+        _userService = new Mock<IUserService>();
+        _authentication = new AuthenticationService(_userService.Object);
+    }
+
+    [Fact]
+    public async Task Authenticate_WithUsername_ReturnUser()
+    {
+        var givenUser = UserLoginDTOMockData.GetUserWithUsername();
+        var user = UserMockData.GetUser();
+        _userService.Setup(service => service.GetByUsernameAsync(givenUser.UsernameOrEmail)).Returns(Task.FromResult<User?>(user));
```

Comment:
add a line before Setup. try to use ReturnAsync and get rid of Task.FromResult

Reply:
Is the Setup part of the Act ? Because i thought it is part of the Arrange

Reply:
this is Arrange, but it does not mean that there should not be a line. Arrange, Act and Assert should be split by comments